src/client/mod.rs

```diff
@@ -103,6 +117,53 @@ pub struct ClientResponse {
     version: Option<http::Version>,
 }
 
+#[derive(Debug)]
+pub struct RequestHeader {
+    pub key: String,
+    pub value: String,
+}
+
+#[derive(Debug)]
+pub struct ClientRequestBuilder<T>
```

Comment:
Don't need to reinvent a common struct. You can use [`http::request::Request`](https://docs.rs/http/0.2.1/http/request/struct.Request.html) here, which embraces builder pattern.

Maybe we can simply add a method in ClientExt that accepts a [`http::request::Request`](https://docs.rs/http/0.2.1/http/request/struct.Request.html)? Can it be a better solution?

Reply:
My good, this look so nice, I will try it !